src/Features/Core/Portable/ConvertToInterpolatedString/AbstractConvertConcatenationToInterpolatedStringRefactoringProvider.cs

```diff
@@ -130,29 +137,47 @@ private Task<Document> UpdateDocumentAsync(Document document, SyntaxNode root, S
                         // not:
                         //      {InterpolatedStringText}{Interpolation}{InterpolatedStringText}{InterpolatedStringText}
                         var existingInterpolatedStringTextNode = content.Last();
-                        var newText = ConcatinateTextToTextNode(generator, existingInterpolatedStringTextNode, textWithoutQuotes);
+                        var newText = ConcatenateTextToTextNode(generator, existingInterpolatedStringTextNode, stringText);
                         content[content.Count - 1] = newText;
                     }
                     else
                     {
                         // This is either the first string literal we have encountered or it is the most recent one we've seen
                         // after adding an interpolation.  Add a new interpolated-string-text-node to the list.
-                        content.Add(generator.InterpolatedStringText(generator.InterpolatedStringTextToken(textWithoutQuotes)));
+                        content.Add(generator.InterpolatedStringText(generator.InterpolatedStringTextToken(stringText)));
                     }
                 }
                 else
                 {
                     content.Add(generator.Interpolation(piece.WithoutTrivia()));
                 }
                 // Update this variable to be true every time we encounter a new string literal expression
-                // so we know to concatinate future string literals together if we encounter them.
-                previousContentWasStringLiteralExpression = currentContentIsStringLiteral;
+                // so we know to concatenate future string literals together if we encounter them.
+                previousContentWasStringLiteralExpression = currentContentIsStringLiteral || currentContentIsInterpolatedStringLiteral;
```

Comment:
is this right? existing code looks liek to concentrate "a" + "b", not $"a {1}" + "b"